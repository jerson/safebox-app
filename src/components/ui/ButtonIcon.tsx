import React, { useRef } from 'react';
import { StyleProp, TextStyle } from 'react-native';
import Touchable, { TouchableProps } from './Touchable';
import Icon from 'react-native-vector-icons/Feather';

export interface ButtonIconProps extends TouchableProps {
  name: string;
  iconStyle?: StyleProp<TextStyle>;
}

function ButtonIcon(props: ButtonIconProps) {
  const touchableRef = useRef(null);

  const { name, onPress, iconStyle, ...extraProps } = props;
  return (
    <Touchable ref={touchableRef} {...extraProps} onPress={onPress}>
      <Icon name={name} style={iconStyle} />
    </Touchable>
  );
}
export default ButtonIcon;
