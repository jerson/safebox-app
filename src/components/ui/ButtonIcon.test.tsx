import React from 'react';
import * as renderer from 'react-test-renderer';
import ButtonIcon, { ButtonIconProps } from './ButtonIcon';

const props: ButtonIconProps = {
  iconStyle: { fontSize: 20 },
  style: {
    width: 30,
    height: 30,
    alignItems: 'center',
    justifyContent: 'center'
  },
  name: 'plus-circle',
  onPress: jest.fn()
};

test('renders without crashing', () => {
  const snapshot = renderer.create(<ButtonIcon {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<ButtonIcon {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
