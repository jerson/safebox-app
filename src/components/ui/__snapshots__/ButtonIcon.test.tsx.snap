// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 30,
      "justifyContent": "center",
      "opacity": 1,
      "width": 30,
    }
  }
  testID="touchable"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "fontSize": 20,
        },
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï†µ
  </Text>
</_class>
`;
